packages/manager/modules/pci/src/projects/project/storages/databases/database/service-integration/add/add.controller.js

```diff
@@ -6,47 +9,196 @@ export default class {
+              curr.destinationEngine,
+            )
+          ) {
+            existingCapability.destinationEngines.push(curr.destinationEngine);
+          }
+        } else {
+          const capabilityArray = {
+            type: curr.type,
+            parameters: curr.parameters,
+            sourceEngines: [curr.sourceEngine],
+            destinationEngines: [curr.destinationEngine],
+          };
+          acc.push(capabilityArray);
+        }
+        return acc;
+      },
+      [],
+    );
   }
 
   cancel() {
-    this.trackDashboard(`service_integration::add_${this.engineName}_cancel`);
+    this.trackDashboard(
+      `service_integration::add_${this.database.engine}_cancel`,
+    );
     this.goBack();
   }
 
+  handleIntegrationChange() {
+    const { sourceEngines, destinationEngines } = this.integrationCapability;
+    // Filter services list according to integration capability
+    if (sourceEngines === destinationEngines) {
+      const filteredServices = this.servicesList.filter((db) =>
+        sourceEngines.includes(db.engine),
+      );
+      this.sourceServiceList = filteredServices;
+      this.destinationServiceList = filteredServices;
+    } else {
+      this.sourceServiceList =
+        isEqual(sourceEngines, [this.database.engine]) &&
+        !destinationEngines.includes(this.database.engine)
+          ? [this.database]
+          : this.servicesList.filter((db) => sourceEngines.includes(db.engine));
+      this.destinationServiceList =
+        isEqual(destinationEngines, [this.database.engine]) &&
+        !sourceEngines.includes(this.database.engine)
+          ? [this.database]
+          : this.servicesList.filter((db) =>
+              destinationEngines.includes(db.engine),
+            );
+    }
+    // Preselect service if only one element
+    this.sourceService =
+      this.sourceServiceList.length === 1 ? this.sourceServiceList[0] : null;
+    this.destinationService =
+      this.destinationServiceList.length === 1
+        ? this.destinationServiceList[0]
+        : null;
+    // Emtpy parameters array
+    this.parameters = {};
+
+    // Check if preselected data is valid
+    this.checkIntegrationValidity();
+  }
+
+  get isDisabledSourceSelect() {
+    if (!this.sourceServiceList || this.sourceServiceList.length === 0) {
+      return true;
+    }
+    return (
+      this.integrationCapability.sourceEngines.includes(this.database.engine) &&
+      !this.integrationCapability.destinationEngines.includes(
+        this.database.engine,
+      )
+    );
+  }
+
+  get isDisabledDestinationSelect() {
+    if (
+      !this.destinationServiceList ||
+      this.destinationServiceList.length === 0
+    ) {
+      return true;
+    }
+    return (
+      this.integrationCapability.destinationEngines.includes(
+        this.database.engine,
+      ) &&
+      !this.integrationCapability.sourceEngines.includes(this.database.engine)
+    );
+  }
```

Comment:
**nitpick:** These 2 functions can be combined since they have a common logic.

Reply:
Done here: https://github.com/ovh/manager/pull/7186/commits/1deaf0646c832b5ecaf9fc76964d5ebb35a75b3a